### ISSUE TITLE
Custom endpoint configuration not retrieved - JWT validation

### ISSUE THREAD

**Issue by CarloGarcia** [arcus-azure/arcus.webapi#227]
I have encountered an issue where my B2C JWT tokens are failing to be validated.

It seems that the custom endpoint (openIdConnectDiscoveryUri) is never queried and,  as a result, the signing keys for my endpoint is never retrieved.

I suspect that the code inside this block is never reached, is it? The validation parameters can never be null (due to constructer validation right?) and as a result I believe is causing the issue being reported here.

https://github.com/arcus-azure/arcus.webapi/blob/383fbaa1462b8ce3b80df77fb47e639210cac81c/src/Arcus.WebApi.Security/Authorization/Jwt/JwtTokenReader.cs#L271

**Comment by stijnmoreels** [comment]
This is not entirely correct. Only in some cases is the token validation parameters initialized. It's only when the consumer didn't specified these parameters that the section you're referring to gets executed.
This reader cam be created in many different configurations.

**Comment by CarloGarcia** [comment]
I am certain that I have missed something (since this is the first time that I have used this package and I am still getting to know it) however I can't see a way to create a reader for a custom OpenID endpoint without specifying validation parameters. 

I can see what you mean, but those reader configurations you mentioned only seem to be for cases where the default OpenID endpoint is used, which is not the case in my scenario.

Can you provide an example snippet of how I can construct a reader with a custom OpenID endpoint and without validation parameters?

**Comment by stijnmoreels** [comment]
As far as I know, it makes sense that a custom OpenID endpoint requires custom validation parameters as we can't guess how this endpoint will validate it's requests. So, if you want to use this custom OpenID endpoint functionality, you should specify how this custom endpoint will authorize requests.

Adding @gverstraete bc he was involved in the proposal.

**Comment by gverstraete** [comment]
Yes indeed, I created the proposal for this, but I wanted to use it solely for the validation a MSI token. I'm kind of unaware of the extensions added after that... But in essence I think you can pass the variables you want to check? - What do you assume as a basic set Carlo?

**Comment by stijnmoreels** [comment]
Is this still a problem @CarloGarcia ? Or do you have found something?
